apps/api/src/persistence/database/arango-database-for-collection.ts

```diff
@@ -6,20 +7,28 @@ import { InternalError } from '../../lib/errors/InternalError';
 import { Maybe } from '../../lib/types/maybe';
 import { DeepPartial } from '../../types/DeepPartial';
 import { ArangoDatabase } from './arango-database';
-import { ArangoCollectionId } from './collection-references/ArangoCollectionId';
-import { ArangoDatabaseDocument } from './utilities/mapEntityDTOToDatabaseDTO';
+import { ArangoDatabaseDocument } from './utilities/mapEntityDTOToDatabaseDocument';
 
 /**
  * Note that at this level we are working with a `DatabaseDocument` (has _key
  * and _id), not an `EntityDTO`. The mapping is taken care of in the
  * repositories layer.
  */
 export class ArangoDatabaseForCollection<TEntity extends HasAggregateId> {
-    #collectionID: ArangoCollectionId;
+    #collectionID: string;
 
     #arangoDatabase: ArangoDatabase;
 
-    constructor(arangoDatabase: ArangoDatabase, collectionName: ArangoCollectionId) {
+    /**
+     * We used to type `collectionName: ArangoCollectionName` and enforce this
+     * statically. We found this led to unwanted coupling and unnecessary
+     * complexity. All collections should be created dynamically if missing
+     * after being registered dynamically via annotations.
+     *
+     * At the db implementation level, we can simply check if the collection
```

Comment:
Let's make sure we can dynamically register arango collection IDs in the near future.